### ISSUE TITLE
Vl53L1X library distance measurement problem

### ISSUE THREAD

**Issue by Autobot86** [sparkfun/SparkFun_VL53L1X_Arduino_Library#1]
hai friends,
First i grate thanks to sparkfun team for providing this library with arduino.
I tested VL53L1X sensor with Arduino board using this library i have some problems for measurement distance.
sometime measurement not work zero value come many time and struck this code then i reset it's work normally.
this code only measure after 30cm short distance not cover (example: 0 to 30cm reading not correct).
so please help me for remove this bugs on code.

**Comment by nseidle** [comment]
Hi!

Please let us know the basics:

* What platform are you using?
* How is the board wired to your platform?

Please see section 3.3 of the [VL53L1X datasheet](https://cdn.sparkfun.com/assets/b/f/2/9/8/VL53L1X_Datasheet.pdf). 4cm is the minimum distance so you should be able to measure 30cm.

Your issue may be related to section 2.5.1. The library currently only supports long range sensing. We still need to implement the short and medium ranges.

**Comment by Autobot86** [comment]
Thanks for replay,
I used VL53L1X sensor in STMicroelectronics board and interface with Arduino nano board.
test code on below 
 
#include <Wire.h>

#include "SparkFun_VL53L1X_Arduino_Library.h"
VL53L1X distanceSensor;

void setup(void)
{
  Wire.begin();

  Serial.begin(9600);
  Serial.println("VL53L1X Qwiic Test");

  if (distanceSensor.begin() == false)
    Serial.println("Sensor offline!");
    
  if (distanceSensor.begin() == true)
    Serial.println("Sensor online!");
}

void loop(void)
{
  distanceSensor.startMeasurement(); //Write configuration bytes to initiate measurement

  //Poll for completion of measurement. Takes 40-50ms.
  while (distanceSensor.newDataReady() == false)
  delay(10);

  uint16_t distance = distanceSensor.getDistance(); //Get the result of the measurement from the sensor

  Serial.print("Distance(mm): ");
  Serial.print(distance);
}

sensor some time hanging stuck the code no data come in the serial, After reset Arduino nano the data come few seconds then hanging.

**Comment by nseidle** [comment]
Closing due to inactivity. Please re-open or let us know how you fixed your problem!

**Comment by ArashShFard** [comment]
I am seeing a similar problem with the SparkFun distance sensor board. The sensor measures 0mm up to 120mm of actual distance (where it measures 1mm). Then, the measured distance is inaccurate up to 300mm of actual distance. From there on, the measurements are accurate. 

This issue does not appear to be related to the distance mode. I have tried all the three distance modes and am still getting the same results.

I have now switched to this [library](https://github.com/pololu/vl53l1x-st-api-arduino) which does not have the same issue and measurements are accurate from 40mm and on.

Further, the issue can be easily duplicated using the examples in this library and will disappear in the examples from the alternative library.
